pkg/api/v1/helper/helpers_test.go

```diff
@@ -26,6 +26,56 @@ import (
 	"k8s.io/apimachinery/pkg/labels"
 )
 
+func TestIsOpaqueIntResourceName(t *testing.T) {
+	testCases := []struct {
+		resourceName v1.ResourceName
+		expectVal    bool
+	}{
+		{
+			resourceName: "pod.alpha.kubernetes.io/opaque-int-resource-foo",
+			expectVal:    true,
+		},
+		{
+			resourceName: "foo",
+			expectVal:    false,
+		},
+		{
+			resourceName: "",
+			expectVal:    false,
+		},
+	}
+
+	for i, tc := range testCases {
+		v := IsOpaqueIntResourceName(tc.resourceName)
+		if v != tc.expectVal {
+			t.Errorf("[%v]expected %v but got %v.", i, tc.expectVal, v)
```

Comment:
I'd recommend including the input value instead of the index to denote the test case in order to save the user from having to manually map indices to cases. Admittedly not a huge problem when you only have a handful of test cases, but still one extra step and hop to the code base you need to do to figure out which test failed.

I might also consider using Go sub-tests to streamline specifying the test name and parallelize execution of the test cases, just as good practice.

Finally, I'd swap the expected/got output order in the message to match the assertion order. (Not sure if there's anything more specific in the Kubernetes style guide, but Go prefers the [_got first, want second_](https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures) order.)

Comments apply to the other new test too. :-)